workers/task-driver/src/driver.rs

```diff
@@ -260,19 +256,24 @@ impl TaskExecutor {
             }
         }
 
-        // Pause the queues for the affected local wallets
-        for wallet_id in wallet_ids.iter() {
-            self.state().pause_task_queue(wallet_id)?;
-        }
-
-        // Start the task optimistically assuming that the queues are paused
-        self.add_preemptive_task(task_id);
-        let preemptive_tasks = self.preemptive_tasks.clone();
-
         let fut = self.create_task_future(true /* immediate */, task_id, task);
+        let preemptive_tasks = self.preemptive_tasks.clone();
         let state = self.state().clone();
         self.runtime.spawn(
             async move {
+                // Pause the queues for the affected local wallets
+                for wallet_id in wallet_ids.iter() {
+                    state
+                        .pause_task_queue(wallet_id)
+                        .expect("error proposing task queue pause for wallet {wallet_id}")
+                        .await
+                        .expect("error pausing task queue for wallet {wallet_id}");
+                }
+
+                // Add the task to the preemptive tasks list so that notification requests can
+                // be registered for it
+                preemptive_tasks.write().unwrap().insert(task_id);
```

Comment:
now that there's a decent chunk of logic in this block, probably makes sense to break this out into its own `async` method, easier to follow that way